core/tests/test_api.py

```diff
@@ -0,0 +1,234 @@
+    page_2_abs_url = get_abs_url(url, page=2)
+    page_3_abs_url = get_abs_url(url, page=3)
+
+    assert page_1_response.status_code == 200
+    assert page_2_response.status_code == 200
+    assert page_3_response.status_code == 200
+    assert len(page_1_response.data["results"]) == PAGE_SIZE
+    assert len(page_2_response.data["results"]) == PAGE_SIZE
+    assert len(page_3_response.data["results"]) == 1
+    assert page_1_response.data["previous"] is None
+    assert page_1_response.data["next"] == page_2_abs_url
+    assert page_2_response.data["previous"] in (endpoint_abs_url, page_1_abs_url)
+    assert page_2_response.data["next"] == page_3_abs_url
+    assert page_3_response.data["previous"] == page_2_abs_url
+    assert page_3_response.data["next"] is None
+    assert page_1_response.data["count"] == 2 * PAGE_SIZE + 1
+    assert page_2_response.data["count"] == 2 * PAGE_SIZE + 1
+    assert page_3_response.data["count"] == 2 * PAGE_SIZE + 1
+
+
+@pytest.mark.parametrize("privacy", visible_privacies)
+def test_api_communities_list_three_pages(
+    client: Client, create_communities: CreateCommunitiesFixture, privacy: str
+) -> None:
+    create_communities(count=PAGE_SIZE * 2 + 1, posts_per_community=0, privacy=privacy)
+    url = reverse("api-communities-list")
+    three_pages_pagination(url, client)
+
+
+@pytest.mark.parametrize("privacy", visible_privacies)
+def test_api_posts_three_pages(client: Client, create_communities: CreateCommunitiesFixture, privacy: str) -> None:
+    create_communities(count=1, posts_per_community=2 * PAGE_SIZE + 1, privacy=privacy)
+    url = reverse("api-posts-list-view")
+    three_pages_pagination(url, client)
+
+
+def test_api_communities_list_private(client: Client, create_communities: CreateCommunitiesFixture) -> None:
+    create_communities(count=10, posts_per_community=2, privacy=Community.PRIVATE)
+    response = client.get(reverse("api-communities-list"))
+    assert response.status_code == 200
+    assert response.data["results"] == []
+
+
+@pytest.mark.parametrize("privacy", visible_privacies)
+def test_api_community_posts_three_pages(
+    client: Client, create_communities: CreateCommunitiesFixture, privacy: str
+) -> None:
+    tested_community = create_communities(count=1, posts_per_community=2 * PAGE_SIZE + 1, privacy=privacy)[0]
+    url = reverse("api-communities-posts-list", kwargs={"slug": tested_community.slug})
+    three_pages_pagination(url, client)
+
+
+@pytest.mark.parametrize("privacy", visible_privacies)
+def test_api_communities_list_ten(client: Client, create_communities: CreateCommunitiesFixture, privacy: str) -> None:
+    create_communities(count=10, posts_per_community=0, privacy=privacy)
+    response = client.get(reverse("api-communities-list"))
+    assert Community.objects.count() == 10
+    assert Post.objects.count() == 0
+    assert response.status_code == 200
+    results = response.data["results"]
+    assert isinstance(results, list)
```

Comment:
I don't think this is assert needed. It should be an iterable, if it's not it will fail on next line.